### ISSUE TITLE
MaxListenersExceededWarning in Node 8

### ISSUE THREAD

**Issue by earllapura** [tgriesser/knex#3050]
# Environment

Knex version: 0.16.3
Database + version: SQLite3 4.0.6
OS: Windows 10
Node version: 8.9.4

# Bug
I tried using `knex` instance on two integration tests with migrations and seeding, each with `beforeEach` hooks as:
```javascript
const knex = require('../../db/knex')
//...
    beforeEach((done) => {
        knex.initialize()
        knex.migrate.rollback().then(() => {
            knex.migrate.latest().then(() => {
                return knex.seed.run().then(() => {
                    done()
                })
            })
        })
    })
    afterEach((done) => {
        knex.migrate.rollback().then(() => {
            knex.destroy().then(() => {
                done()
            })
        })
    })
```
with `knex.js` as:
```javascript
var environment = process.env.NODE_ENV || 'development';
var config = require('../knexfile.js')[environment];

module.exports = require('knex')(config);
```
The problem is I get the `MaxListenersExceededWarning` at the migration process, which seems to  with stack as below:
```
(node:6600) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 start listeners added. Use emitter.setMaxListeners() to increase limit
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 start listeners added. Use emitter.setMaxListeners() to increase limit
    at _addListener (events.js:280:19)
    at Client_SQLite3.addListener (events.js:297:10)
    at redefineProperties (C:\projects\scope-api\node_modules\knex\lib\util\make-knex.js:225:15)
    at Function.withUserParams (C:\projects\scope-api\node_modules\knex\lib\util\make-knex.js:116:7)
    at new Migrator (C:\projects\scope-api\node_modules\knex\lib\migrate\Migrator.js:55:24)
    at Function.get (C:\projects\scope-api\node_modules\knex\lib\util\make-knex.js:192:16)
    at Context.beforeEach (C:\projects\scope-api\tests\integration\scopemodel.test.js:10:14)
    at callFnAsync (C:\projects\scope-api\node_modules\mocha\lib\runnable.js:400:21)
    at Hook.Runnable.run (C:\projects\scope-api\node_modules\mocha\lib\runnable.js:342:7)
    at next (C:\projects\scope-api\node_modules\mocha\lib\runner.js:317:10)
    at Immediate._onImmediate (C:\projects\scope-api\node_modules\mocha\lib\runner.js:347:5)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
```
It seems that running `require('knex')(config)` might be replicating the listeners. The [test](https://github.com/tgriesser/knex/blob/e413805561294f069edb3781a1f16f1f63469e9d/test/unit/knex.js#L67) associated with this is using `knex.withUserParams` which might seem to not duplicate the listeners.
Any help on this?

**Comment by kibertoad** [comment]
Try using 0.16.4-next,should help.

**Comment by earllapura** [comment]
Oh, that fix was not yet released. That explains it.

**Comment by shuangq** [comment]
I met the same problem even using 0.16.4-next1.  I use `knex.migrate.rollback()` before each test. As the number of tests grows, Node throws the same warnings.
```
(node:87779) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 start listeners added. Use emitter.setMaxListeners() to increase limit
    at _addListener (events.js:246:17)
    at Client_PG.addListener (events.js:262:10)
    at redefineProperties (/[project]/node_modules/knex/lib/util/make-knex.js:225:15)
    at Function.withUserParams (/[project]/node_modules/knex/lib/util/make-knex.js:116:7)
    at new Migrator (/[project]/node_modules/knex/lib/migrate/Migrator.js:55:24)
    at Function.get (/[project]/node_modules/knex/lib/util/make-knex.js:192:16)
    at knex.migrate.rollback.then (/[project]/tests/routes/index.test.js:25:30)
    at tryCatcher (/[project]/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/[project]/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/[project]/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/[project]/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/[project]/node_modules/bluebird/js/release/promise.js:694:18)
    at _drainQueueStep (/[project]/node_modules/bluebird/js/release/async.js:138:12)
    at _drainQueue (/[project]/node_modules/bluebird/js/release/async.js:131:9)
    at Async._drainQueues (/[project]/node_modules/bluebird/js/release/async.js:147:5)
    at Immediate.Async.drainQueues (/[project]/node_modules/bluebird/js/release/async.js:17:14)
    at processImmediate (timers.js:632:19)
    at process.topLevelDomainCallback (domain.js:120:23)
```
Any help?

**Comment by kibertoad** [comment]
It was not in next1 for some reason, master only. We'll release soon.

**Comment by kibertoad** [comment]
Fixed in 0.16.4-next2
